Sources/Plasma/CoreLib/hsMatrix44.h

```diff
@@ -163,9 +175,77 @@ struct hsMatrix44 {
 
     static hsMatrix44 mult_fpu(const hsMatrix44& a, const hsMatrix44& b);
     static hsMatrix44 mult_sse3(const hsMatrix44& a, const hsMatrix44& b);
+#ifdef HS_BUILD_FOR_APPLE
+    static hsMatrix44 mult_accelerate(const hsMatrix44 &a, const hsMatrix44 &b);
+#endif
 };
 
 ST_DECL_FORMAT_TYPE(const hsMatrix44&);
 
+static inline hsMatrix44 IMatrixMul34(const hsMatrix44& lhs, const hsMatrix44& rhs)
+{
+    hsMatrix44 ret;
+    ret.NotIdentity();
+    
+#ifdef HS_BUILD_FOR_APPLE
+    vDSP_mmul((const float*)lhs.fMap, 1, (const float*)rhs.fMap, 1, (float*)&(ret.fMap), 1, 3, 4, 4);
+#else
+    ret.fMap[0][0] = lhs.fMap[0][0] * rhs.fMap[0][0]
+        + lhs.fMap[0][1] * rhs.fMap[1][0]
+        + lhs.fMap[0][2] * rhs.fMap[2][0];
+
+    ret.fMap[0][1] = lhs.fMap[0][0] * rhs.fMap[0][1]
+        + lhs.fMap[0][1] * rhs.fMap[1][1]
+        + lhs.fMap[0][2] * rhs.fMap[2][1];
+
+    ret.fMap[0][2] = lhs.fMap[0][0] * rhs.fMap[0][2]
+        + lhs.fMap[0][1] * rhs.fMap[1][2]
+        + lhs.fMap[0][2] * rhs.fMap[2][2];
+
+    ret.fMap[0][3] = lhs.fMap[0][0] * rhs.fMap[0][3]
+        + lhs.fMap[0][1] * rhs.fMap[1][3]
+        + lhs.fMap[0][2] * rhs.fMap[2][3]
+        + lhs.fMap[0][3];
+
+    ret.fMap[1][0] = lhs.fMap[1][0] * rhs.fMap[0][0]
+        + lhs.fMap[1][1] * rhs.fMap[1][0]
+        + lhs.fMap[1][2] * rhs.fMap[2][0];
+
+    ret.fMap[1][1] = lhs.fMap[1][0] * rhs.fMap[0][1]
+        + lhs.fMap[1][1] * rhs.fMap[1][1]
+        + lhs.fMap[1][2] * rhs.fMap[2][1];
+
+    ret.fMap[1][2] = lhs.fMap[1][0] * rhs.fMap[0][2]
+        + lhs.fMap[1][1] * rhs.fMap[1][2]
+        + lhs.fMap[1][2] * rhs.fMap[2][2];
+
+    ret.fMap[1][3] = lhs.fMap[1][0] * rhs.fMap[0][3]
+        + lhs.fMap[1][1] * rhs.fMap[1][3]
+        + lhs.fMap[1][2] * rhs.fMap[2][3]
+        + lhs.fMap[1][3];
+
+    ret.fMap[2][0] = lhs.fMap[2][0] * rhs.fMap[0][0]
+        + lhs.fMap[2][1] * rhs.fMap[1][0]
+        + lhs.fMap[2][2] * rhs.fMap[2][0];
+
+    ret.fMap[2][1] = lhs.fMap[2][0] * rhs.fMap[0][1]
+        + lhs.fMap[2][1] * rhs.fMap[1][1]
+        + lhs.fMap[2][2] * rhs.fMap[2][1];
+
+    ret.fMap[2][2] = lhs.fMap[2][0] * rhs.fMap[0][2]
+        + lhs.fMap[2][1] * rhs.fMap[1][2]
+        + lhs.fMap[2][2] * rhs.fMap[2][2];
+
+    ret.fMap[2][3] = lhs.fMap[2][0] * rhs.fMap[0][3]
+        + lhs.fMap[2][1] * rhs.fMap[1][3]
+        + lhs.fMap[2][2] * rhs.fMap[2][3]
+        + lhs.fMap[2][3];
+#endif
+    ret.fMap[3][0] = ret.fMap[3][1] = ret.fMap[3][2] = 0;
+    ret.fMap[3][3] = 1.f;
+
+    return ret;
+}
```

Comment:
I was thinking more of an opt-in header than this. As it stands, the symbols from `Accelerate/Accelerate.h` will leak into the whole engine pretty easily.

Reply:
Lemme take a look at making this more robust